### PR TITLE
feat: retrievable deals

### PR DESCRIPTION
**Pull request by bajtos**
Create a new table `retrievable_deals` and insert 10k `(cid, expires_at)` pairs to bootstrap the dataset.

Links:
- https://www.notion.so/pl-strflt/Moving-IPNI-queries-to-Spark-checkers-c58175c0f1c841e6bb82ac44a1472a98
- https://github.com/filecoin-station/spark/issues/40

### CODE REVIEW

File: migrations/039.do.retrievable-deals.sql [hunk 1]
```diff
@@ -0,0 +1,10013 @@
+CREATE TABLE retrievable_deals (
+  cid TEXT NOT NULL PRIMARY KEY,
+  expires_at TIMESTAMPTZ NOT NULL
+);
```
Comment (bajtos) [inline migrations/039.do.retrievable-deals.sql:4]: @juliangruber do you have any feedback on the schema definition and the table name in particular?

Quoting from the design doc:

> Introduce a new SQL table with a list of FIL+ LDN deals we want to test for retrievability. This table will be similar to `retrieval_templates` and have the following columns:
>     - cid (primary key; extracted from deal Label)
>     - expires_at (timestamp with tz, calculated from deal proposal’s EndEpoch field)
>     
> In the future, we will need metadata to match IPNI responses to this particular deal. We don’t have that implemented yet, so let's keep the db schema simple and the table size small for now.
>
> I considered adding `id` as a primary key, but we don’t use it anywhere. We can rework the primary key to include multiple columns after we add more columns. The benefit of the proposed scheme is that we will naturally enforce the uniqueness of templates. (The rows in `retrieval_templates` are not unique.)

Comment (juliangruber) [reply migrations/039.do.retrievable-deals.sql:4]: `retrievable_deals` makes sense to me! The only alternative I can think of is something like `ipni_responses`

File: migrations/039.do.retrievable-deals.sql [hunk 2]
```diff
@@ -0,0 +1,10013 @@
...
+
+-- First 10k deals to bootstrap the dataset. Ingest the remaining 22m deals manually.
+
+TRUNCATE TABLE retrievable_deals;
```
Comment (juliangruber) [inline migrations/039.do.retrievable-deals.sql:8]: why truncate it first?

**Review by juliangruber — APPROVED** [review]

File: migrations/039.do.retrievable-deals.sql (hunk 2 above)
Comment (bajtos) [reply migrations/039.do.retrievable-deals.sql:8]: Ah, I copy&pasted the SQL generated by fil-deal-ingester. We don't need to truncate the table in this migration script, I'll remove that statement.

**Pull request merged**